### PR TITLE
Add version string to sysimg triple

### LINKED ISSUE #51562: ld: warning: no platform load command found in sys-o.a assuming macOS
I assume this is minor given it correctly assumes macOS

```
Output ──────  22.320461 seconds
    LINK usr/lib/julia/sys.dylib
ld: warning: no platform load command found in '/Users/ian/Documents/GitHub/julia/usr/lib/julia/sys-o.a[2](text#0.o)', assuming: macOS
ld: warning: no platform load command found in '/Users/ian/Documents/GitHub/julia/usr/lib/julia/sys-o.a[3](text#1.o)', assuming: macOS
ld: warning: no platform load command found in '/Users/ian/Documents/GitHub/julia/usr/lib/julia/sys-o.a[4](text#2.o)', assuming: macOS
ld: warning: no platform load command found in '/Users/ian/Documents/GitHub/julia/usr/lib/julia/sys-o.a[5](metadata.o)', assuming: macOS
ld: warning: no platform load command found in '/Users/ian/Documents/GitHub/julia/usr/lib/julia/sys-o.a[6](sysimg.o)', assuming: macOS
    JULIA stdlib/Unicode.release.image
```

macOS Sonoma
```
julia> versioninfo()
Julia Version 1.11.0-DEV.593
Commit f2d1276be8 (2023-10-03 12:04 UTC)
Platform Info:
  OS: macOS (arm64-apple-darwin23.0.0)
  CPU: 10 × Apple M2 Pro
  WORD_SIZE: 64
  LLVM: libLLVM-15.0.7 (ORCJIT, apple-m1)
  Threads: 1 on 6 virtual cores
```

### PR DESCRIPTION
**Pull request by gbaraldi**
Maybe we want to change this to something more dynamic, but this at least avoids the warning

Fixes https://github.com/JuliaLang/julia/issues/51562

### CODE REVIEW

File: src/aotcompile.cpp [hunk 1]
```diff
@@ -1500,7 +1500,13 @@ void jl_dump_native_impl(void *native_code,
...
     } else if (TheTriple.isOSDarwin()) {
         TheTriple.setObjectFormat(Triple::MachO);
-        TheTriple.setOS(llvm::Triple::MacOSX);
+        SmallString<16> Str;
+        Str += "macosx";
+        if (TheTriple.isAArch64())
+            Str += "11.0.0"; // Update this if MACOSX_VERSION_MIN changes
+        else
+            Str += "10.14.0";
+        TheTriple.setOSName(Str);
```
Comment (vtjnash) [inline src/aotcompile.cpp:1509]:
```suggestion
        TheTriple.setOSName("darwin18"); // set -mmacosx-version-min=10.14, like Make.inc
```

Comment (gbaraldi) [reply src/aotcompile.cpp:1509]: That is not what clang does

Comment (gbaraldi) [reply src/aotcompile.cpp:1509]:
```
Apple clang version 15.0.0 (clang-1500.0.40.1)
Target: arm64-apple-darwin23.0.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
 "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang" -cc1 -triple arm64-apple-macosx14.0.0
```

Comment (gbaraldi) [reply src/aotcompile.cpp:1509]: https://github.com/llvm/llvm-project/blob/a5f446bc4bb1ac78d6852cc8e251a1229899b783/clang/lib/Driver/ToolChains/Darwin.cpp#L1049-L1062

File: src/aotcompile.cpp [hunk 2]
```diff
@@ -1500,7 +1500,13 @@ void jl_dump_native_impl(void *native_code,
         TheTriple.setObjectFormat(Triple::COFF);
     } else if (TheTriple.isOSDarwin()) {
         TheTriple.setObjectFormat(Triple::MachO);
-        TheTriple.setOS(llvm::Triple::MacOSX);
+        SmallString<16> Str;
+        Str += "macosx";
```
[the remark this answers is not in the document]
Comment (gbaraldi) [reply src/aotcompile.cpp:1504]: Annoyingly `ld64` still expects it

File: src/aotcompile.cpp (hunk 1 above)
Comment (vtjnash) [reply src/aotcompile.cpp:1509]: Oh, interesting to see how the `-triple` and `-target` values disagree there. Alright, I guess this seems right then, assuming it still triggers the right codepaths in LLVM.

Comment (gbaraldi) [reply src/aotcompile.cpp:1509]: It does generate the LC_BUILD_VERSION command, which makes `ld` happier

File: src/aotcompile.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (giordano) [reply src/aotcompile.cpp:1504]: I learnt it the hard way: https://github.com/JuliaPackaging/BinaryBuilderBase.jl/issues/285

**Pull request merged**